### PR TITLE
Added pre-commit hooks and linters

### LINKED ISSUE #48: [Feature Request] Adding pre-commit hooks
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

Addition of pre-commit hooks for linters and code formatting.

### Use Case

Code optimisation

### Record

- [X] I have read the Contributing Guidelines
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by sank8-2**
Please review for any changes
Closes #48

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v5.0.0
+    hooks:
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+      - id: requirements-txt-fixer
```
Comment (ahaapple) [inline .pre-commit-config.yaml:7]: we don't have python language，So I think we don't need the `requirements-txt-fixer`.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  - repo: https://github.com/mgedmin/check-manifest
```
Comment (ahaapple) [inline .pre-commit-config.yaml:9]: we don't have python language，do we need it?

File: .gitignore [hunk 3]
```diff
@@ -1 +1,3 @@
 scripts
+node_modules
```
Comment (ahaapple) [inline .gitignore:2]: Why do we need these ignores?

File: .pre-commit-config.yaml (hunk 2 above)
Comment (sank8-2) [reply .pre-commit-config.yaml:9]: Ok I'll remove it

File: .gitignore (hunk 3 above)
Comment (sank8-2) [reply .gitignore:2]: For packages of husky

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    rev: "0.49"
+    hooks:
+      - id: check-manifest
+        stages: [manual]
+
+  - repo: local
+    hooks:
+      - id: prettier-staged
+        name: Prettier Staged
+        entry: git-format-staged -f 'prettier --ignore-unknown --stdin --stdin-filepath "$FILENAME"'
```
Comment (ahaapple) [inline .pre-commit-config.yaml:19]: Does this require each contributor to install git-format-staged and prettier on their local machine in advance?

Comment (sank8-2) [reply .pre-commit-config.yaml:18]: It is included in the package.json of the root directory so it gets installed from there

Comment (sank8-2) [reply .pre-commit-config.yaml:18]: Should I mention it in the README under Contributing?

Comment (ahaapple) [reply .pre-commit-config.yaml:18]: Hi，I would like to confirm, after having this PR, if the developer clones the repository, can he directly git commit the code without performing any additional operations?

Comment (sank8-2) [reply .pre-commit-config.yaml]: I added installation command for it in the latest commit

Comment (sank8-2) [reply .pre-commit-config.yaml]: Yes, with the update he can

**Review by ahaapple — APPROVED** [review]

**Pull request merged**